src/appender/index_writer.rs

```diff
@@ -0,0 +1,92 @@
+use anyhow::{anyhow, ensure, Result};
+use kanal::{bounded, Receiver, Sender};
+use std::{mem::size_of, slice::from_raw_parts, thread};
+
+use crate::{
+    formats::log::IndexFileHeader,
+    util::{BlockIODevice, LogItem},
+};
+
+#[derive(Debug)]
+pub(crate) struct IndexWriter<T> {
+    sender: Sender<Vec<T>>,
+}
+
+impl<T: LogItem> IndexWriter<T> {
+    pub(crate) fn new(file: impl BlockIODevice, expected_header: IndexFileHeader) -> Result<Self> {
+        let file_size = file.len()?;
+        let header_len = size_of::<IndexFileHeader>() as u64;
+        if file_size > 0 {
+            ensure!(
+                file_size > header_len && (file_size - header_len) % size_of::<T>() as u64 == 0,
+                "Invalid log index file"
```

Comment:
instead of invalid using broken, and don't panic, try to recover for example mv file to file.old and recover the file